src/mds/Server.cc

```diff
@@ -7108,6 +7108,12 @@ void Server::handle_client_link(MDRequestRef& mdr)
   if (target_pin != dir->inode &&
       target_realm->get_subvolume_ino() !=
       dir->inode->find_snaprealm()->get_subvolume_ino()) {
+    if (target_pin->get_parent_inode()->is_stray()) {
+      target_pin->add_waiter(CInode::WAIT_UNLINK,
+                             new C_MDS_RetryRequest(mdcache, mdr));
```

Comment:
We should use the `get_projected_parent_dir()` instead:

```
-    if (target_pin->get_parent_inode()->is_stray()) {
+    if (target_pin->get_projected_parent_dir()->inode->is_stray()) {
```

Reply:
right!